idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/annotators/coercion/coercionAnnotatorExplicitProperty.kt

```diff
@@ -0,0 +1,72 @@
+package arrow.meta.ide.plugins.proofs.annotators.coercion
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.plugins.proofs.annotators.addLocalQuickFix
+import arrow.meta.ide.plugins.proofs.annotators.addProblemDescriptor
+import arrow.meta.ide.plugins.proofs.annotators.implicitProofAnnotatorTextAttributes
+import arrow.meta.ide.plugins.proofs.explicit
+import arrow.meta.ide.plugins.proofs.isCoerced
+import arrow.meta.ide.plugins.proofs.participatingTypes
+import arrow.meta.phases.CompilerContext
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.plugins.proofs.phases.coerceProof
+import com.intellij.codeInsight.daemon.impl.quickfix.GoToSymbolFix
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.psi.PsiElement
+import org.celtric.kotlin.html.body
+import org.celtric.kotlin.html.html
+import org.celtric.kotlin.html.text
+import org.jetbrains.kotlin.idea.KotlinQuickDocumentationProvider
+import org.jetbrains.kotlin.js.resolve.diagnostics.findPsi
+import org.jetbrains.kotlin.psi.KtNamedDeclaration
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameSafe
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+val IdeMetaPlugin.coercionAnnotatorExplicitProperty: ExtensionPhase
+  get() = addAnnotator(
+    annotator = Annotator { element: PsiElement, holder: AnnotationHolder ->
+      val ctx = element.project.getService(CompilerContext::class.java)
+      element.safeAs<KtProperty>()
+        ?.takeIf { ctx.isCoerced(it) }
+        ?.let { ktProperty: KtProperty ->
+          ktProperty.delegateExpressionOrInitializer?.let {
+            ktProperty.participatingTypes()?.let { (subtype, supertype) ->
+              ctx.coerceProof(subtype, supertype)?.let { proof ->
+                proof.through.findPsi()?.let { proofPsi ->
```

Comment:
nit: 
```suggestion
                proof.through.findPsi().safeAs<KtNamedDeclaration>?.let { proofPsi ->
```
plus removing the cast in line 63

Reply:
@danimontoya ready to merge once this is addressed